Fix for Amasty filter "Availability" on product grid. By default this feature used old Magento inventory from table "cataloginventory_stock_item" and it's doesn't work with MSI.
This patch will change used table to "inventory_stock_2" where stock for MSI is stored and it's a temporary solution because Amasty works on adjust module for MSI.
Issue appeared in ticket: IPET-1069

diff --git a/Plugin/Catalog/Ui/DataProvider/Product/ProductDataProvider.php b/Plugin/Catalog/Ui/DataProvider/Product/ProductDataProvider.php
index 1d6707d..6dbb4cc 100644
--- a/Plugin/Catalog/Ui/DataProvider/Product/ProductDataProvider.php
+++ b/Plugin/Catalog/Ui/DataProvider/Product/ProductDataProvider.php
@@ -227,14 +227,24 @@ class ProductDataProvider
 
     private function addInventoryColumn(Collection $collection, string $amastyColumnName, string $columnName): void
     {
+// BEGIN PATCH
+//        $collection->joinField(
+//            $amastyColumnName,
+//            'cataloginventory_stock_item',
+//            $columnName,
+//            'product_id=entity_id',
+//            '{{table}}.stock_id=1',
+//            'left'
+//        );
         $collection->joinField(
             $amastyColumnName,
-            'cataloginventory_stock_item',
+            'inventory_stock_2',
             $columnName,
-            'product_id=entity_id',
-            '{{table}}.stock_id=1',
+            'sku=sku',
+            null,
             'left'
         );
+// END PATCH
     }
 
     protected function _addLowStock(Collection $collection): void
diff --git a/Ui/Component/Listing/Column/Availability.php b/Ui/Component/Listing/Column/Availability.php
index e32fe3e..1c6a685 100644
--- a/Ui/Component/Listing/Column/Availability.php
+++ b/Ui/Component/Listing/Column/Availability.php
@@ -36,7 +36,9 @@ class Availability extends Column implements OptionSourceInterface
     public function getOptionArray()
     {
         return [
-            self::DISABLE_MANAGE_STOCK => __('Manage Stock Disabled'),
+            // BEGIN PATCH
+            //self::DISABLE_MANAGE_STOCK => __('Manage Stock Disabled'),
+            // END PATCH
             self::IN_STOCK => __('In Stock'),
             self::OUT_OF_STOCK => __('Out Of Stock')
         ];
@@ -71,10 +73,13 @@ class Availability extends Column implements OptionSourceInterface
      */
     public function getAvailableExpression()
     {
-        $configManageStock = $this->stockConfiguration->getManageStock();
-
-        return 'IF(((at_amasty_availability.manage_stock = 0 AND at_amasty_availability.use_config_manage_stock = 0)' .
-            ' OR (' . $configManageStock . ' = 0 AND at_amasty_availability.use_config_manage_stock = 1)),'
-            . self::DISABLE_MANAGE_STOCK . ', at_amasty_availability.is_in_stock)';
+// BEGIN PATCH
+        return 'at_amasty_availability.is_salable';
+//        $configManageStock = $this->stockConfiguration->getManageStock();
+//
+//        return 'IF(((at_amasty_availability.manage_stock = 0 AND at_amasty_availability.use_config_manage_stock = 0)' .
+//            ' OR (' . $configManageStock . ' = 0 AND at_amasty_availability.use_config_manage_stock = 1)),'
+//            . self::DISABLE_MANAGE_STOCK . ', at_amasty_availability.is_in_stock)';
+// END PATCH
     }
 }
diff --git a/Ui/DataProvider/Product/AddAvailabilityFilterToCollection.php b/Ui/DataProvider/Product/AddAvailabilityFilterToCollection.php
index 5a0409e..dae08fd 100644
--- a/Ui/DataProvider/Product/AddAvailabilityFilterToCollection.php
+++ b/Ui/DataProvider/Product/AddAvailabilityFilterToCollection.php
@@ -31,15 +31,24 @@ class AddAvailabilityFilterToCollection implements AddFilterToCollectionInterfac
         if ($collection->getFlag('amasty_instock_filter')) {
             return;
         }
-
+// BEGIN PATCH
+//        $collection->joinField(
+//            'amasty_availability',
+//            'cataloginventory_stock_item',
+//            $this->availabilityColumn->getAvailableExpression(),
+//            'product_id=entity_id',
+//            '{{table}}.stock_id=1',
+//            'left'
+//        );
         $collection->joinField(
             'amasty_availability',
-            'cataloginventory_stock_item',
+            'inventory_stock_2',
             $this->availabilityColumn->getAvailableExpression(),
-            'product_id=entity_id',
-            '{{table}}.stock_id=1',
+            'sku=sku',
+            null,
             'left'
         );
+// END PATCH
 
         $collection->getSelect()->where($this->availabilityColumn->getAvailableExpression() . '= ?', $condition['eq']);
         $collection->setFlag('amasty_instock_filter', 1);
diff --git a/view/adminhtml/ui_component/product_listing.xml b/view/adminhtml/ui_component/product_listing.xml
index 35c8981..9323a55 100644
--- a/view/adminhtml/ui_component/product_listing.xml
+++ b/view/adminhtml/ui_component/product_listing.xml
@@ -338,18 +338,20 @@
                 </item>
             </argument>
         </column>
-        <column name="amasty_low_stock">
-            <argument name="data" xsi:type="array">
-                <item name="options" xsi:type="object">Amasty\Pgrid\Model\Product\Lowstock</item>
-                <item name="config" xsi:type="array">
-                    <item name="component" xsi:type="string">Amasty_Pgrid/js/grid/columns/select</item>
-                    <item name="dataType" xsi:type="string">select</item>
-                    <item name="label" xsi:type="string" translate="true">Low Stock</item>
-                    <item name="amastyExtra" xsi:type="boolean">true</item>
-                    <item name="visible" xsi:type="boolean">false</item>
-                </item>
-            </argument>
-        </column>
+<!-- BEGIN PATCH -->
+<!--        <column name="amasty_low_stock">-->
+<!--            <argument name="data" xsi:type="array">-->
+<!--                <item name="options" xsi:type="object">Amasty\Pgrid\Model\Product\Lowstock</item>-->
+<!--                <item name="config" xsi:type="array">-->
+<!--                    <item name="component" xsi:type="string">Amasty_Pgrid/js/grid/columns/select</item>-->
+<!--                    <item name="dataType" xsi:type="string">select</item>-->
+<!--                    <item name="label" xsi:type="string" translate="true">Low Stock</item>-->
+<!--                    <item name="amastyExtra" xsi:type="boolean">true</item>-->
+<!--                    <item name="visible" xsi:type="boolean">false</item>-->
+<!--                </item>-->
+<!--            </argument>-->
+<!--        </column>-->
+<!-- END PATCH -->
         <column name="amasty_tier_price">
             <argument name="data" xsi:type="array">
                 <item name="config" xsi:type="array">
@@ -363,19 +365,21 @@
                 </item>
             </argument>
         </column>
-        <column name="amasty_backorders">
-            <argument name="data" xsi:type="array">
-                <item name="config" xsi:type="array">
-                    <item name="label" xsi:type="string" translate="true">Backorders</item>
-                    <item name="editor" xsi:type="string">select</item>
-                    <item name="dataType" xsi:type="string">select</item>
-                    <item name="component" xsi:type="string">Amasty_Pgrid/js/grid/columns/select</item>
-                    <item name="amastyExtra" xsi:type="boolean">true</item>
-                    <item name="visible" xsi:type="boolean">false</item>
-                </item>
-                <item name="options" xsi:type="object">Magento\CatalogInventory\Model\Source\Backorders</item>
-            </argument>
-        </column>
+<!-- BEGIN PATCH -->
+<!--        <column name="amasty_backorders">-->
+<!--            <argument name="data" xsi:type="array">-->
+<!--                <item name="config" xsi:type="array">-->
+<!--                    <item name="label" xsi:type="string" translate="true">Backorders</item>-->
+<!--                    <item name="editor" xsi:type="string">select</item>-->
+<!--                    <item name="dataType" xsi:type="string">select</item>-->
+<!--                    <item name="component" xsi:type="string">Amasty_Pgrid/js/grid/columns/select</item>-->
+<!--                    <item name="amastyExtra" xsi:type="boolean">true</item>-->
+<!--                    <item name="visible" xsi:type="boolean">false</item>-->
+<!--                </item>-->
+<!--                <item name="options" xsi:type="object">Magento\CatalogInventory\Model\Source\Backorders</item>-->
+<!--            </argument>-->
+<!--        </column>-->
+<!-- END PATCH -->
         <column name="weight">
             <argument name="data" xsi:type="array">
                 <item name="config" xsi:type="array">
@@ -391,17 +395,19 @@
                 </item>
             </argument>
         </column>
-        <column name="amasty_qty_sold">
-            <argument name="data" xsi:type="array">
-                <item name="config" xsi:type="array">
-                    <item name="filter" xsi:type="string">textRange</item>
-                    <item name="label" xsi:type="string" translate="true">Qty Sold</item>
-                    <item name="amastyExtra" xsi:type="boolean">true</item>
-                    <item name="visible" xsi:type="boolean">false</item>
-                    <item name="component" xsi:type="string">Amasty_Pgrid/js/grid/columns/column</item>
-                </item>
-            </argument>
-        </column>
+<!-- BEGIN PATCH -->
+<!--        <column name="amasty_qty_sold">-->
+<!--            <argument name="data" xsi:type="array">-->
+<!--                <item name="config" xsi:type="array">-->
+<!--                    <item name="filter" xsi:type="string">textRange</item>-->
+<!--                    <item name="label" xsi:type="string" translate="true">Qty Sold</item>-->
+<!--                    <item name="amastyExtra" xsi:type="boolean">true</item>-->
+<!--                    <item name="visible" xsi:type="boolean">false</item>-->
+<!--                    <item name="component" xsi:type="string">Amasty_Pgrid/js/grid/columns/column</item>-->
+<!--                </item>-->
+<!--            </argument>-->
+<!--        </column>-->
+        <!-- END PATCH -->
     </columns>
     <container name="pgrid_edit_sources_modal">
         <argument name="data" xsi:type="array">
